components/uaepass-authenticator/src/main/java/org/wso2/carbon/identity/uaepass/authenticator/UAEPassAuthenticator.java

```diff
@@ -0,0 +1,1109 @@
+import org.apache.oltu.oauth2.common.exception.OAuthProblemException;
+import org.apache.oltu.oauth2.common.exception.OAuthSystemException;
+import org.apache.oltu.oauth2.common.message.types.GrantType;
+import org.wso2.carbon.base.MultitenantConstants;
+import org.wso2.carbon.identity.application.authentication.framework.AbstractApplicationAuthenticator;
+import org.wso2.carbon.identity.application.authentication.framework.FederatedApplicationAuthenticator;
+import org.wso2.carbon.identity.application.authentication.framework.context.AuthenticationContext;
+import org.wso2.carbon.identity.application.authentication.framework.exception.AuthenticationFailedException;
+import org.wso2.carbon.identity.application.authentication.framework.exception.LogoutFailedException;
+import org.wso2.carbon.identity.application.authentication.framework.model.AuthenticatedUser;
+import org.wso2.carbon.identity.application.authentication.framework.util.FrameworkUtils;
+import org.wso2.carbon.identity.application.common.model.ClaimMapping;
+import org.wso2.carbon.identity.application.common.model.Property;
+import org.wso2.carbon.identity.application.common.util.IdentityApplicationConstants;
+import org.wso2.carbon.identity.base.IdentityConstants;
+import org.wso2.carbon.identity.claim.metadata.mgt.exception.ClaimMetadataException;
+import org.wso2.carbon.identity.claim.metadata.mgt.model.ExternalClaim;
+import org.wso2.carbon.identity.core.ServiceURLBuilder;
+import org.wso2.carbon.identity.core.URLBuilderException;
+import org.wso2.carbon.identity.core.util.IdentityCoreConstants;
+import org.wso2.carbon.identity.core.util.IdentityTenantUtil;
+import org.wso2.carbon.identity.core.util.IdentityUtil;
+import org.wso2.carbon.identity.uaepass.authenticator.exception.UAEPassAuthnFailedException;
+import org.wso2.carbon.identity.uaepass.authenticator.exception.UAEPassUserInfoFailedException;
+import org.wso2.carbon.identity.uaepass.authenticator.internal.UAEPassDataHolder;
+import org.wso2.carbon.user.api.UserRealm;
+import org.wso2.carbon.user.api.UserStoreException;
+import org.wso2.carbon.user.core.UserStoreManager;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.UnsupportedEncodingException;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.text.ParseException;
+
+import java.util.Map;
+import java.util.HashMap;
+import java.util.List;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.Set;
+import java.util.HashSet;
+
+public class UAEPassAuthenticator extends AbstractApplicationAuthenticator
+        implements FederatedApplicationAuthenticator {
+
+    private static final Log LOG = LogFactory.getLog(UAEPassAuthenticator.class);
+
+    /**
+     * Checks wheather the request and response can handle by the federated IdP.
```

Comment:
Grammar mistakes. Please check other occurrences as well.
```suggestion
     * Checks whether the request and response can handle by the federated IdP.
```

Reply:
Fixed with : [bf5488f](https://github.com/wso2-extensions/identity-outbound-auth-uaepass/pull/1/commits/bf5488f695a7cae8933ece11d28fcab07cd8e53f)